libc/string_lenchr.c

```diff
@@ -150,9 +154,291 @@ TEST(string_len, big)
 }
 
 
+/*
+////////////////////////////////////////////////////////////////////////////////////
+*/
+
+
+TEST_SETUP(string_chr)
+{
+}
+
+
+TEST_TEAR_DOWN(string_chr)
+{
+}
+
+
+TEST(string_chr, basic)
+{
+	const char *lorem = "Lorem ipsum dolor sit amet, consectetur adipiscing elit y";
+	int sz;
+
+	sz = strlen(lorem);
+
+	/* Check of proper working base */
+	TEST_ASSERT_EQUAL_STRING(lorem, strchr(lorem, 'L'));
+	TEST_ASSERT_EQUAL_STRING(lorem, strrchr(lorem, 'L'));
+	TEST_ASSERT_EQUAL_STRING(lorem, memchr(lorem, 'L', sz));
+
+	TEST_ASSERT_EQUAL_STRING("lor sit amet, consectetur adipiscing elit y", strchr(lorem, 'l'));
+	TEST_ASSERT_EQUAL_STRING("lit y", strrchr(lorem, 'l'));
+	TEST_ASSERT_EQUAL_STRING("lor sit amet, consectetur adipiscing elit y", memchr(lorem, 'l', sz));
+
+	TEST_ASSERT_EQUAL_STRING(NULL, strchr(lorem, 'x'));
+	TEST_ASSERT_EQUAL_STRING(NULL, strrchr(lorem, 'x'));
+	TEST_ASSERT_EQUAL_STRING(NULL, memchr(lorem, 'x', sz));
+
+	TEST_ASSERT_EQUAL_STRING(&lorem[sz], strchr(lorem, lorem[sz]));
+	TEST_ASSERT_EQUAL_STRING(&lorem[sz], strrchr(lorem, lorem[sz]));
+	TEST_ASSERT_EQUAL_STRING(&lorem[sz], memchr(lorem, lorem[sz], sz + 1));
+}
+
+
+TEST(string_chr, big)
+{
+	char str[PATH_MAX] = { 0 };
+
+	/* Long string case */
+	memset(str, 'a', PATH_MAX - 1);
+	str[PATH_MAX - 5] = 'b';
+
+	TEST_ASSERT_EQUAL_STRING("baaa", strchr(str, 'b'));
+	TEST_ASSERT_EQUAL_STRING("baaa", strrchr(str, 'b'));
+	TEST_ASSERT_EQUAL_STRING("baaa", memchr(str, 'b', sizeof(str)));
+
+	TEST_ASSERT_EQUAL_STRING(str, strchr(str, 'a'));
+	TEST_ASSERT_EQUAL_STRING("a", strrchr(str, 'a'));
+	TEST_ASSERT_EQUAL_STRING(str, memchr(str, 'a', sizeof(str)));
+
+	TEST_ASSERT_EQUAL_STRING(NULL, strchr(str, 'x'));
+	TEST_ASSERT_EQUAL_STRING(NULL, strrchr(str, 'x'));
+	TEST_ASSERT_EQUAL_STRING(NULL, memchr(str, 'x', sizeof(str)));
+}
+
+
+TEST(string_chr, special)
+{
+	const char *exp = "Ipsum",
+			   *strWhites = "Lorem \n \t \e \r \b \v \f \\ Ipsum",
+			   *strMod = "Lorem %d/%f/%x/%i/%c/%s Ipsum";
```

Comment:
(strWhites) You have now the separate test case for white spaces, so it's not needed there. Above that special characters do not include white spaces, so it's not the place for it, we want to test only special characters there.

(strMod) Modifiers are nothing more than a combination of a % (one of the special characters, which are tested in this test case) and a letter (tested in the `basic` test case). That's why there is no need to test `strMod`. Modifiers are specific for functions like `scanf` or `printf` and then the behavior for different modifiers should be tested.

```suggestion
```